File: Daily/2023/744_findSmallestLetterGreaterThanTarget.py
from typing import *


class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        n = len(letters)
        left, right = 0, n - 1

        while left < right:
            mid = (left + right) // 2
            if ord(letters[mid]) <= ord(target):
                left = mid + 1
            else:
                right = mid

        if ord(letters[left]) <= ord(target):
            return letters[0]

        return letters[left]

File: Daily/2023/744_findSmallestLetterGreaterThanTarget.cs
public class Solution {
    public char NextGreatestLetter(char[] letters, char target) {
        int n = letters.Length;
        int left = 0, right = n - 1;

        while (left < right) 
        {
            int mid = (left + right) / 2;
            if ( (int)letters[mid] <= (int)target ) 
            {
                left = mid + 1;
            }
            else
            {
                right = mid;
            }
        }

        return (int)letters[left] <= (int)target ? letters[0] : letters[left];
    }
}